### PR TITLE
Lukasp/testsuite improvements

### PR DESCRIPTION
**Pull request by LukasPukenis**
Testsuite has now the ability to catch libdrop's API errors. Also fixed testrunner locking issue

### CODE REVIEW

File: test/drop_test/ffi.py [hunk 1]
```diff
@@ -480,27 +497,33 @@ def start(self, addr: str, dbpath: str, max_reqs_per_sec: int):
...
         return ctypes.string_at(version).decode("utf-8")
 
     def __del__(self):
         err = self._lib.norddrop_destroy(self._instance)
         if err != 0:
             err_type = LibResult(err).name
-            raise Exception(f"norddrop_destory has failed with code: {err}({err_type})")
+            raise DropException(
+                f"norddrop_destory has failed with code: {err}({err_type})", err
+            )
```
Comment (matszczygiel) [inline test/drop_test/ffi.py:526]: Is it OK to throw in the destructor in Python?

File: test/scenarios.py [hunk 2]
```diff
@@ -6766,6 +6766,132 @@
             ),
         },
     ),
+    Scenario(
+        "scenario31-3",
+        "Remove file on sending side without rejecting it first. Expect error and file present in JSON output",
```
Comment (matszczygiel) [inline test/scenarios.py:6771]: Can you clarify the description here that it's about removing the file from the DB, not physically from the disk?

File: test/drop_test/ffi.py (hunk 1 above)
Comment (LukasPukenis) [reply test/drop_test/ffi.py:526]: Honestly, no idea didn't even think about it. Quick googling doesn't reveal anything as well. Do you have any idea about possible side effects?

File: test/drop_test/action.py [hunk 3]
```diff
@@ -545,9 +552,8 @@ def __init__(self, uuid_slot: int, fid):
         self._fid = fid
 
     async def run(self, drop: ffi.Drop):
-        UUIDS_LOCK.acquire()
-        drop.remove_transfer_file(UUIDS[self._uuid_slot], self._fid)
-        UUIDS_LOCK.release()
+        with UUIDS_LOCK:
```
Comment (matszczygiel) [inline test/drop_test/action.py:555]: Me like it :+1:

**Review by matszczygiel — APPROVED** [review]
Nice improvement :+1:

File: test/drop_test/action.py (hunk 3 above)
Comment (LukasPukenis) [reply test/drop_test/action.py:555]: That was actually the thing that was blocking me when I mentioned it. A ton of `print()` later I found it :D

File: test/drop_test/ffi.py (hunk 1 above)
Comment (matszczygiel) [reply test/drop_test/ffi.py:526]: No, but I'm not that deep into Python. In Rust or C++ it's a severe offense

File: test/scenarios.py (hunk 2 above)
Comment (LukasPukenis) [reply test/scenarios.py]: Sure! I've clarified that it's about removal from the transfer itself

File: test/drop_test/ffi.py (hunk 1 above)
Comment (LukasPukenis) [reply test/drop_test/ffi.py:526]: I also have no idea about python, can't really find anything. I thought about maybe doing the creation and destructor explicitly but that doesn't seem worth it to me as that's only done once in the app lifecycle

Comment (matszczygiel) [reply test/drop_test/ffi.py:526]: Yeah, so let's leave it as is. We will continue the discussion when it breaks in any way

**Review by matszczygiel — DISMISSED** [review]

File: test/scenarios.py [hunk 4]
```diff
@@ -6766,6 +6766,257 @@
...
+                            {
+                                event.File(
+                                    FILES["testfile-small"].id,
+                                    "testfile-small",
+                                    1048576,
+                                ),
+                            },
+                        )
+                    ),
+                    action.Sleep(2),
```
Comment (matszczygiel) [inline test/scenarios.py:6789]: When do we need to sleep here?

Comment (LukasPukenis) [reply test/scenarios.py]: Removed

**Review by matszczygiel — APPROVED** [review]

**Pull request merged**